File: Bouchonnois.Tests/Builders/CommandBuilder.cs
using Bouchonnois.Domain.Démarrer;

namespace Bouchonnois.Tests.Builders
{
    public class CommandBuilder
    {
        private (string, int)[] _chasseurs = Array.Empty<(string, int)>();
        private int _nbGalinettes;

        public static CommandBuilder DémarrerUnePartieDeChasse() => new();

        public CommandBuilder Avec(params (string, int)[] chasseurs)
        {
            _chasseurs = chasseurs;
            return this;
        }

        public CommandBuilder SurUnTerrainRicheEnGalinettes(int nbGalinettes = 3)
        {
            _nbGalinettes = nbGalinettes;
            return this;
        }

        public DemarrerPartieDeChasse Build()
            => new(
                new TerrainDeChasse("Pitibon sur Sauldre", _nbGalinettes),
                _chasseurs.Select(c => new Chasseur(c.Item1, c.Item2))
            );
    }
}

File: java/src/test/java/bouchonnois/tests/builders/CommandBuilder.java
package bouchonnois.tests.builders;

import io.vavr.Tuple;
import io.vavr.Tuple2;
import lombok.Getter;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class CommandBuilder {
    @Getter
    private List<Tuple2<String, Integer>> chasseurs = new ArrayList<>();
    private int nbGalinettes;

    public static CommandBuilder demarrerUnePartieDeChasse() {
        return new CommandBuilder();
    }

    @SuppressWarnings("varargs")
    @SafeVarargs
    public final CommandBuilder avec(Tuple2<String, Integer>... chasseurs) {
        Collections.addAll(this.chasseurs, chasseurs);
        return this;
    }

    public CommandBuilder surUnTerrainRicheEnGalinettes() {
        this.nbGalinettes = 3;
        return this;
    }

    public Tuple2<String, Integer> getTerrain() {
        return Tuple.of("Pitibon sur Sauldre", nbGalinettes);
    }
}
